### DISCUSSION TITLE
Not able to generate JPG file

### DISCUSSION THREAD

**Discussion by shub823** [ImageMagick/ImageMagick#6569] [Help]
### ImageMagick version

7.1.0-49

### Operating system, version and so on

Windows 2016 64 bit

### Description

We are trying to merge two image (please see the below commands) and generate a single image final image using Java Code.

**Command**: "magick "Y:\Test_CleanArt_16x9_3840x2160.jpg" "Y:\Test_LOGO_es_419_16x9_3840x2160.png" -resize 3840x2160\! -flatten -strip "D:\Share\layering_output_images\18_08_2023_03\layering_277921\Test_123456.PNG""

**Error** : "magick: must specify image size `\Test_CleanArt_16x9_3840x2160.jpg' @ error/raw.c/ReadRAWImage/140"

We are running this command via Java code on Wildfly and facing this issue. But the same command works perfectly if we run it manually on cmd.

And suggestions are welcome.

Thanks

**Comment by snibgo** [comment]
Your error message refers to a different directory and file than your command. Perhaps you have made a copy-paste mistake.

The error says it is trying to read a raw image. Your command is reading files with JPG and PNG extensions. Perhaps one of those files is bad.

**Reply by shub823** [reply]
Hi Snigbo, You are correct, it is a copy paste issue. Can you pleases see the updated error. 

Also when I try running the command explicity on the cmd , it runs very well without any error, but the same is not happenning via Java Code and giving me the mentioned errors.

Looking forward for your help.

**Comment by fmw42** [comment]
You have not shown the updated error or it is still the same issue.   You have not shown any java code for us to analyze.

> Error : "magick: must specify image size `\Test_CleanArt_16x9_3840x2160.jpg' @ error/raw.c/ReadRAWImage/140"

This says that your so called jpg image is actually a raw image, not jpg, (or the jpg header is missing or corrupt) and IM wants to read it but can't because a raw image must have its size specified.

**Reply by shub823** [reply]
Yes,that is the updated error and the same command is running successfully without any issues on cmd if we try it manually. But not via code.

**Java Code**: 
                        String cmd="magick "Y:\Test_CleanArt_16x9_3840x2160.jpg" "Y:\Test_LOGO_es_419_16x9_3840x2160.png" -resize 
                                      3840x2160! -flatten -strip "D:\Share\layering_output_images\18_08_2023_03\layering_277921\Test_123456.PNG"";
                            long start = Calendar.getInstance().getTime().getTime();
                            Process runtime = Runtime.getRuntime().exec(cmd);
                            logger.info(" Layering Stared -- " + start);
                            runtime.waitFor(10, TimeUnit.SECONDS);
                            long end = Calendar.getInstance().getTime().getTime();
                            logger.info("Layering Over  " + end);
                            logger.info("Layering Time lapse  " + (start - end));
                            logger.info("Layering Process Info - " + runtime.info());
                            logger.info("Layering Process is Alive - " + runtime.isAlive());
                            logger.info("Layering Process Exit Value - " + runtime.exitValue());
                            InputStream errorStream = runtime.getErrorStream();
                            int c = 0;
                            StringBuilder sb = new StringBuilder();
                            logger.info("Layering Process error started ----------------- ");
                            while ((c = errorStream.read()) != -1) {
                                sb.append((char) c);
                            }
                            logger.info("Layering Process error ended ------------------- " + sb.toString());
                        
                 This code is running on Wildfly server.

**Comment by snibgo** [comment]
The error message refers to:
```
`\Test_CleanArt_16x9_3840x2160.jpg'
```
Note the leading backslash. Filenames should not start with backslash. So `magick` is being passed a bad command.

Your Java has:
```
String cmd="magick "Y:\Test_CleanArt_16x9_3840x2160.jpg" "Y:\Test_LOGO_es_419_16x9_3840x2160.png" -resize
3840x2160! -flatten -strip "D:\Share\layering_output_images\18_08_2023_03\layering_277921\Test_123456.PNG";
```
There are 7 double-quotes, but there should be an even number, like 6 or 8.

When double-quotes are nested inside double-quotes, they usually need to be escaped. I don't know Java, and can't advise.

**Reply by fmw42** [reply]
Perhaps you can use single outer quotes for the whole cmd and double quotes for the file names or vice-versa?

As snibgo said, if you have a starting quote before magick, then you need another ending quote at the end of whole command. You only have one around the beginning and ending of the output name right now.
